### ISSUE TITLE
Add the ability to read a recommendation

### ISSUE THREAD

**Issue by amateurPotato** [NYUDevOps2021-recommendation/recommendations#7]
**As a** recommendation service consumer
**I need** the ability to read recommendation for a product
**So that** I can search for alternative products 

**Details & Assumptions:** 
* The search request is for a single productId
* The productId may or may not be present in database
* The productId may or may not have recommendation of type cross-sell, upsell, related-products
* The recommendations will have lists of product for all three categories of recommendation: upsell, cross-sell, related-products

**Acceptance Criteria:**

```gherkin
Given [a productId]
When [I search for recommendations with productId]
Then [I get list of recommended productIds for each category upsell, cross-sell, related-products]
```
** API Contract: **

Input: 
```
Method: GET
API URI: /recommendations/
PathParameters: productId
```
Output:
```
ResponseCode: 200
Response: { } or 
                   { 
                       upsell: [list of productIds], 
                       crossSell: [list of productIds], 
                       relatedProducts : [list of productIds
                    }
```
Error Handling:
ResponseCode | Description
--- | --- 
400 | no product found for the given productId
401 | not an authorized user
500 | internal server error while searching for recommendation

**Comment by JinweiDai** [comment]
Maybe "productId" should not include a big letter?

**Comment by JinweiDai** [comment]
Should we add another PathParameters named {relation} that contains upsell or crossSell or relatedProducts. Instead of return all related prodauct now ?

**Comment by amateurPotato** [comment]
> Should we add another PathParameters named {relation} that contains upsell or crossSell or relatedProducts. Instead of return all related product now ?

@JinweiDai  - We can have a separate read - with query in path parameters, we can groom  a new story  later, before we start the sprint related to Read stories.

**Comment by amateurPotato** [comment]
> Maybe "productId" should not include a big letter?

@JinweiDai - Here is good guidelines reference:
https://api.gov.au/standards/national_api_standards/naming-conventions.html

I personally don't use '_' in any url components. Let's brainstorm on our api guidelines and best practices this sprint. We can brainstorm on this in our upcoming call.
